src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -861,108 +768,106 @@ public static void PrelinkAll(Type c)
-        // native memory block to it.
-        //====================================================================
+        /// <summary>
+        /// Creates a new instance of "structuretype" and marshals data from a
+        /// native memory block to it.
+        /// </summary>
         public static object PtrToStructure(IntPtr ptr, Type structureType)
         {
-            if (ptr == IntPtr.Zero) return null;
+            if (ptr == IntPtr.Zero)
+            {
+                return null;
+            }
 
             if (structureType == null)
+            {
                 throw new ArgumentNullException(nameof(structureType));
-
+            }
             if (structureType.IsGenericType)
+            {
                 throw new ArgumentException(SR.Argument_NeedNonGenericType, nameof(structureType));
-
-            RuntimeType rt = structureType.UnderlyingSystemType as RuntimeType;
-
-            if (rt == null)
+            }
+            if (!(structureType.UnderlyingSystemType is RuntimeType rt))
+            {
                 throw new ArgumentException(SR.Arg_MustBeType, nameof(structureType));
+            }
 
-            object structure = rt.CreateInstanceDefaultCtor(false /*publicOnly*/, false /*skipCheckThis*/, false /*fillCache*/, true /*wrapExceptions*/);
-            PtrToStructureHelper(ptr, structure, true);
+            object structure = rt.CreateInstanceDefaultCtor(publicOnly: false, skipCheckThis: false, fillCache: false, wrapExceptions: true);
+            PtrToStructureHelper(ptr, structure, allowValueClasses: true);
             return structure;
         }
 
-        public static T PtrToStructure<T>(IntPtr ptr)
-        {
-            return (T)PtrToStructure(ptr, typeof(T));
-        }
+        public static T PtrToStructure<T>(IntPtr ptr) => (T)PtrToStructure(ptr, typeof(T));
 
-        //====================================================================
-        // Helper function to copy a pointer into a preallocated structure.
-        //====================================================================
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        /// <summary>
+        /// Helper function to copy a pointer into a preallocated structure.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern void PtrToStructureHelper(IntPtr ptr, object structure, bool allowValueClasses);
 
-
-        //====================================================================
-        // Freeds all substructures pointed to by the native memory block.
-        // "structureclass" is used to provide layout information.
-        //====================================================================
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        /// <summary>
+        /// Freeds all substructures pointed to by the native memory block.
+        /// "structureclass" is used to provide layout information.
```

Comment:
Is "structureclass" referring to the argument that is now called "structuretype"?